Source/HTTP/httpcall.cpp

```diff
@@ -137,6 +138,91 @@ HRESULT CALLBACK HC_CALL::PerfomAsyncProvider(XAsyncOp op, XAsyncProviderData co
     }
 }
 
+void CALLBACK HC_CALL::CompressRequestBody(void* c, bool canceled)
+{
+    PerformContext* context{ static_cast<PerformContext*>(c) };
+    HC_CALL* call{ context->call };
+
+    if (canceled)
+    {
+        XAsyncComplete(context->asyncBlock, E_ABORT, 0);
+        return;
+    }
+
+    if (call->traceCall)
+    {
+        HC_TRACE_INFORMATION(HTTPCLIENT, "HC_CALL::CompressRequestBody [ID %llu] Iteration %d", TO_ULL(call->id), call->m_iterationNumber);
+    }
+
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+    // Compress body before call if applicable
+    if (call->enableGzipCompression && call->compressionLevel != HCCompressionLevel::None)
```

Comment:
>call->enableGzipCompression && call->compressionLevel != HCCompressionLevel::None [](http://example.com/codeflow?start=8&length=81)

Minor, but I would move this check into the provider above so that CompressRequestBody doesn't even get scheduled if compression is disabled

Reply:
Fixed